tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/BeginMethodHandler.cs

```diff
@@ -42,7 +42,9 @@ static BeginMethodHandler()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static CallTargetState Invoke(TTarget instance)
         {
-            return new CallTargetState(Tracer.Instance.InternalActiveScope, Tracer.Instance.DistributedSpanContext, _invokeDelegate(instance));
+            var activeScope = Tracer.Instance.InternalActiveScope;
+            var spanContextRaw = DistributedTracer.Instance.GetSpanContextRaw() ?? activeScope?.Span?.Context;
```

Comment:
Maybe worth adding a comment at these call sites about why we're _not_ using `Tracer.Instance.DistributedSpanContext`, in case someone tries to optimise it to reduce duplication 🙂

Reply:
done in [0479960](https://github.com/DataDog/dd-trace-dotnet/pull/2857/commits/047996093c53da0580e6caec9914e0ef46835fb6)